arch/rv32i/src/pmp.rs

```diff
@@ -629,8 +633,9 @@ impl<const MAX_AVAILABLE_REGIONS_OVER_TWO: usize> kernel::platform::mpu::KernelM
                             csr::CSR.pmpaddr_set(x * 2, start >> 2);
 
                             // Set access to end address
-                            csr::CSR
-                                .pmpconfig_set(x / 2, cfg_val << 8 | csr::CSR.pmpconfig_get(x / 2));
+                            let new_cfg =
+                                cfg_val << 8 | (csr::CSR.pmpconfig_get(x / 2) & 0xFFFF_00FF);
+                            csr::CSR.pmpconfig_set(x / 2, new_cfg);
```

Comment:
Isn't this what `pmpconfig_modify()` is for?

Reply:
The problem is that `cfg_val` is just a number. This is because it can either hold `csr::pmpconfig::pmpcfg::x1` or `csr::pmpconfig::pmpcfg::x3` values. So to be more flexible we accept any value for `cfg_val` then shift it as required

Reply:
I tried this, and yeah because there are actually 5 fields in those 8 bits, it is very cumbersome to use modify.